File: Algorithm-Swift/Algorithm-Swift/Inflearn/19_분노유발자.swift
//
//  19_분노유발자.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/01.
//

import Foundation

func solution19() {
    let n = Int(readLine()!)!
    let students = readLine()!.split(separator: " ").map { Int($0)! }
    var cnt = 0

    for i in 0..<n-1 {
        var flag = true
        let max = students[i]
        for j in i+1..<n {
            if students[j] >= max {
                flag = false
                break;
            }
        }

        if flag {
            cnt += 1
        }
    }

    print(cnt)
}

File: Algorithm-CPP/19_분노유발자.cpp
#include <cmath>
#include <iostream>
using namespace std;

int main() {
    int n, cnt = 0;
    cin >> n;

    int students[n];
    for(int i = 0; i < n; i++) {
        int height;
        cin >> height;
        students[i] = height;
    }

    for(int i = 0; i < n - 1; i++) {
        bool flag = true;
        int max = students[i];
        for(int j = i + 1; j < n; j++) {
            if(students[j] >= max) {
                flag = false;
                break;
            }
        }

        if(flag) cnt++;
    }

    cout << cnt << endl;

    return 0;
} 